src/supermarket/spec/models/universe_spec.rb

```diff
@@ -4,15 +4,32 @@
   let(:cookbook) { 'redis' }
   let(:version) { '1.3.1' }
 
-  it 'allows you to customize things' do
-    opts = {
-      host: 'narf.example.com',
-      port: 6060,
-      protocol: 'https'
-    }
+  context 'when a host is provided' do
+    it 'allows you to customize things' do
+      opts = {
+        host: 'narf.example.com',
+        port: 6060,
+        protocol: 'https'
+      }
 
-    expect(Universe.protocol_host_port(opts)).to eql('https://narf.example.com:6060')
-    expect(Universe.download_url(cookbook, version, 'https://narf.example.com:6060')).to eql('https://narf.example.com:6060/api/v1/cookbooks/redis/versions/1.3.1/download')
+      expect(Universe.protocol_host_port(opts)).to eql('https://narf.example.com:6060')
+      expect(Universe.download_url(cookbook, version, 'https://narf.example.com:6060')).to eql('https://narf.example.com:6060/api/v1/cookbooks/redis/versions/1.3.1/download')
+    end
+  end
+
+  context 'when a host is not provided' do
+    it 'uses the fqdn environmental variable' do
+      expect(ENV['HOST']).to be_nil
+      ENV['FQDN'] = "localhost"
+
+      opts = {
+        port: 6060,
+        protocol: 'https'
+      }
+
+      expect(Universe.protocol_host_port(opts)).to eql("https://#{ENV['FQDN']}:6060")
+      expect(Universe.download_url(cookbook, version, "https://#{ENV['FQDN']}:6060")).to eql("https://#{ENV['FQDN']}:6060/api/v1/cookbooks/redis/versions/1.3.1/download")
```

Comment:
How would you feel about using a wild non-defaulty string for the FQDN in this test? Like:

```
expected_fqdn = 'this_is_my_supermarket_there_are_many_like_it_but_this_one_is_mine'
ENV['FQDN'] = expected_fqdn

...

expect(Universe.protocol_host_port(opts)).to eql("https://#{expected_fqdn}:6060")
expect(Universe.download_url(cookbook, version, "https://#{expected_fqdn}:6060")).to eql("https://#{expected_fqdn}:6060/api/v1/cookbooks/redis/versions/1.3.1/download")
```

Juuuuuust in case `localhost` sneaks into tests or something as a default.